docs/knowledge-base/DevOps/pipelines.mdx

```diff
@@ -0,0 +1,158 @@
+
+The pipeline can be found [here](https://jenkins.mapcolonies.net/job/Deploy%20Chart%20Polling%20-%20Multibranch/)  
+As you can see in the overview page of the pipeline there are 2 branches which it's monitoring.
+
+![jenkins auto deploy overview](./img/jenkins-auto-deploy-1.png)  
+
+##### How the auto deploy pipeline is working?
+
+* The pipeline is first of all looking at changes introduced in each commit with a `git diff`.
+* After finding the `diff` it'll look at the `deployment.json` in the `scope` (each team's directory in the `helm-charts` repo) directories.
+* Here's an example of `deployment.json` file
+
+```json
+[
+    {
+      "chartLocation": "3d/charts/ingestion",
+      "valuesFile": ["3d/charts/global-values.yaml", "3d/charts/ingestion/ingestion-values.yaml"],
+      "namespace": "3d-dev",
+      "releaseName": "ingestion-3d",
+      "skip": false
+    },
+    {
+      "chartLocation": "3d/charts/serving",
+      "valuesFile": ["3d/charts/global-values.yaml", "3d/charts/serving/serving-values.yaml"],
+      "namespace": "3d-dev",
+      "releaseName": "serving-3d",
+      "skip": true
+    }
+  ]
+```  
+
+* If you wish to skip a deployment you'll need to have `"skip": true` in the chart you wish to skip.
+* When the pipeline will start it'll send a slack message to the `<team>-notifications` channel based on chart location(s) which are going to be deployed.  
+![jenkins auto deploy slack start message](./img/jenkins-auto-deploy-2.png) 
+* After the pipeline will finish it'll send another message to the `<team>-notifications` channel based on chart location(s) which have been deployed with the status of the deployment.  
+
+## Bundler/Unbundler
+
+The bundler and unbundler workflows are intended to help our dev team to bring in the artifacts which were built in the release flow in an easy way to our internal deployments.
+
+We utilize Jenkins for the bundler and unbundler. 
+
+
+![Bundler and Unbundler](./img/bundler-unbundler.png) 
+
+### Bundler pipeline
+
+The [bundler pipeline](https://jenkins.mapcolonies.net/job/bundler/) has 1 parameter which is the `artifacts.json` file.
+
+Here's an example of the `artifacts.json` 
+
+
+```json
+{
+  "docker": {
+    "acrarolibotnonprod.azurecr.io": {
+      "3d-gateway": "v2.4.2",
+      "store-trigger": "v1.5.2"
+    },
+    "docker.io": {
+      "bitnami/redis": "7.2.3",
+      "timonwong/uwsgi-exporter": "latest"
+    }
+  },
+  "helm": {
+    "acrarolibotnonprod.azurecr.io": {
+      "gateway": "2.4.2",
+      "store-trigger": "1.5.2"
+    },
+    "https://charts.bitnami.com/bitnami": {
+      "redis": "18.5.0"
+    }
+  },
+  "git": [
+    "MapColonies/helm-charts"
+  ]
+}
+```  
+
+The bundler pipeline will bundle all the artifacts (docker tar balls, helm packages, git repos) into 1 zip file.
+
+Once the bundler is done it'll send a slack message to the `<team>-notifications` channel based on the user logged in.
```

Comment:
I would have also included where the blob is stored and how to locate it.

Reply:
@michalby24 i think the end user shouldn't be aware as it's not relevant to them as they are getting a pre-signed url